wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/grpc.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import java.util.Base64
+import okhttp3.Headers.Companion.headersOf
+import okio.IOException
+
+/** Returns an exception if the gRPC call didn't have a grpc-status of 0. */
+fun GrpcResponse.grpcResponseToException(suppressed: IOException? = null): IOException? {
```

Comment:
I don’t think I like the signature of this function as a public API.

It’s job is to produce an exception if the response itself is malformed or bad.

Maybe this?

```
fun GrpcResponse.toExceptionIfFailed(suppressed: IOException? = null): IOException? {
}
```